### PR TITLE
Project setup

### PR DESCRIPTION
**Pull request by Mogball**
Project setup and some code for temperature sensors.

Build with CLion or `platformio run`

### CODE REVIEW

**Comment by dhillondeep** [comment]
This setup is fine but if you are using CLion you should follow the CLion instructions
https://docs.teamwaterloop.ca/en/teams/tech/software/embedded.html#creating-platformio-projects
In terms of adding the `embedded-cplusplus` the full repo is fine we can tell CMake to only care about `Wlib` so this should not be too bad. Make a `lib` folder and put the `submodule` in that so that `PlatformIO` can find it

**Comment by dhillondeep** [comment]
This setup is fine but if you are using CLion you should follow the CLion instructions
https://docs.teamwaterloop.ca/en/teams/tech/software/embedded.html#creating-platformio-projects
In terms of adding the `embedded-cplusplus` the full repo is fine we can tell CMake to only care about `Wlib` so this should not be too bad. Make a `lib` folder and put the `submodule` in that so that `PlatformIO` can find it

File: src/register/Register.h [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+#include <Arduino.h>
+#include <Wire.h>
+
+namespace wlp {
+
+    typedef uint16_t size_type;
+
+    class Register {
+        uint8_t m_i2c_addr;
```
Comment (dhillondeep) [inline src/register/Register.h]: I think you should make the Register more generic so like it should not be `i2c_adress` but rather the address of any register. If you wanna make something for I2C we should name it I2C or something so it is more clear

Comment (Mogball) [reply src/register/Register.h]: aight I'll rename it, (I think) it's i2c specfic.

**Comment by Mogball** [comment]
@dhillondeep xd gimme

**Review by dhillondeep — APPROVED** [review]

**Pull request merged**